src/Resolver/DependencyResolver.php

```diff
@@ -177,6 +202,26 @@ public function setContainer(ContainerInterface $container)
     }
 
     /**
+     * @param string $type
+     * @return bool
+     */
+    private function isCallableType(string $type): bool
+    {
+        if ($this->config->isAlias($type)) {
+            $type = $this->config->getClassForAlias($type);
+        }
+
+        if (! class_exists($type) && ! interface_exists($type)) {
+            return false;
+        }
+
+        $reflection = new \ReflectionClass($type);
+
+        return $reflection->hasMethod('__invoke') &&
+               $reflection->getMethod('__invoke')->isPublic();
```

Comment:
Our CS recommends putting conditions on the next line, at one level of indentation:

```php
return $reflection->hasMethod('__invoke')
    && $reflection->getMethod('__invoke')->isPublic();
```